File: backend/app/models/task.py
from sqlalchemy import Column, Integer, String, Text, TIMESTAMP, ForeignKey
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.database import Base

class Task(Base):
    """任务模型"""
    __tablename__ = "tasks"
    
    id = Column(Integer, primary_key=True, index=True)
    name = Column(String(255), nullable=False, comment="任务名称")
    type = Column(String(100), comment="任务类型")
    phase = Column(String(50), comment="任务阶段")
    description = Column(Text, comment="任务描述")
    status = Column(String(50), default='未分配', comment="任务状态")
    create_time = Column(TIMESTAMP, server_default=func.now(), comment="创建时间")
    update_time = Column(TIMESTAMP, server_default=func.now(), onupdate=func.now(), comment="更新时间")
    
    # 关联关系
    assignments = relationship("TaskAssignment", back_populates="task")
    
    def __repr__(self):
        return f"<Task(id={self.id}, name='{self.name}', status='{self.status}')>"

class TaskAssignment(Base):
    """任务分配模型"""
    __tablename__ = "task_assignments"
    
    id = Column(Integer, primary_key=True, index=True)
    task_id = Column(Integer, ForeignKey("tasks.id"), comment="任务ID")
    user_id = Column(Integer, ForeignKey("users.id"), comment="用户ID")
    username = Column(String(255), comment="用户名")
    assigned_at = Column(TIMESTAMP, server_default=func.now(), comment="分配时间")
    status = Column(String(50), default='进行中', comment="执行状态")
    progress = Column(Integer, default=0, comment="进度百分比")
    performance_score = Column(Integer, default=0, comment="性能评分")
    comments = Column(Text, comment="备注")
    last_update = Column(TIMESTAMP, server_default=func.now(), onupdate=func.now(), comment="最后更新")
    
    # 关联关系
    task = relationship("Task", back_populates="assignments")
    user = relationship("User", back_populates="task_assignments")
    
    def __repr__(self):
        return f"<TaskAssignment(id={self.id}, task_id={self.task_id}, username='{self.username}')>" 

File: frontend/src/api/task.ts
import request from '@/utils/request'

export interface Task {
  id: number
  name: string
  type: string
  phase: string
  description: string
  status: string
  create_time: string
  update_time: string
}

export interface TaskAssignment {
  id: number
  task_id: number
  user_id?: number
  username: string
  assigned_at: string
  status: string
  progress: number
  performance_score: number
  comments?: string
  last_update: string
}

export interface TaskWithAssignments extends Task {
  assignments: TaskAssignment[]
}

export interface TaskStats {
  total_tasks: number
  completed_tasks: number
  in_progress_tasks: number
  average_score: number
}

// 获取任务列表
export const getTasks = (params?: {
  skip?: number
  limit?: number
  status?: string
  task_type?: string
}) => {
  return request.get<Task[]>('/tasks', { params })
}

// 获取任务总数
export const getTasksCount = (params?: {
  status?: string
  task_type?: string
}) => {
  return request.get<{ count: number }>('/tasks/count', { params })
}

// 获取任务详情
export const getTask = (taskId: number) => {
  return request.get<TaskWithAssignments>(`/tasks/${taskId}`)
}

// 创建任务
export const createTask = (data: Omit<Task, 'id' | 'create_time' | 'update_time'>) => {
  return request.post<Task>('/tasks', data)
}

// 更新任务
export const updateTask = (taskId: number, data: Partial<Omit<Task, 'id' | 'create_time' | 'update_time'>>) => {
  return request.put<Task>(`/tasks/${taskId}`, data)
}

// 删除任务
export const deleteTask = (taskId: number) => {
  return request.delete(`/tasks/${taskId}`)
}

// 获取任务分配列表
export const getTaskAssignments = (params?: {
  skip?: number
  limit?: number
  status?: string
}) => {
  return request.get<TaskAssignment[]>('/task-assignments', { params })
}

// 获取我的任务
export const getMyTasks = (params?: { status?: string }) => {
  return request.get<TaskAssignment[]>('/my-tasks', { params })
}

// 获取我的任务统计
export const getMyTaskStats = () => {
  return request.get<TaskStats>('/my-task-stats')
}

// 创建任务分配
export const createTaskAssignment = (data: Omit<TaskAssignment, 'id' | 'assigned_at' | 'last_update'>) => {
  return request.post<TaskAssignment>('/task-assignments', data)
}

// 更新任务分配
export const updateTaskAssignment = (assignmentId: number, data: {
  status?: string
  progress?: number
  performance_score?: number
  comments?: string
}) => {
  return request.put<TaskAssignment>(`/task-assignments/${assignmentId}`, data)
}

// 更新我的任务
export const updateMyTask = (assignmentId: number, data: {
  status?: string
  progress?: number
  comments?: string
}) => {
  return request.put<TaskAssignment>(`/my-tasks/${assignmentId}`, data)
} 